File: api/src/main/java/net/impactdev/gts/api/events/PingEvent.java
package net.impactdev.gts.api.events;

import net.impactdev.impactor.api.event.ImpactorEvent;
import org.spongepowered.api.util.annotation.eventgen.GenerateFactoryMethod;

import java.time.Instant;
import java.util.UUID;

@GenerateFactoryMethod
public interface PingEvent extends ImpactorEvent {

    UUID getPingID();

    Instant getTimeSent();

}

File: api/src/main/kotlin/net/impactdev/gts/api/events/PingEvent.kt
package net.impactdev.gts.api.events

import net.impactdev.impactor.api.event.ImpactorEvent
import java.time.Instant
import java.util.*

interface PingEvent : ImpactorEvent {
    @get:Param(0)
    val pingID: UUID?

    @get:Param(1)
    val timeSent: Instant?
}